### DISCUSSION TITLE
Is it possible to get the current request name via script?

### DISCUSSION THREAD

**Discussion by CoenraadS** [usebruno/bruno#954] [Q&A] [answered]
Hi, I'm looking to get the current request name programmatically.

If I console.log the `bru` object, I see a `collectionPath`, but not anything which shows me the current request.

I also tried to log the `req` object, but didn't see it there either.

**Comment by n00o** [comment, marked as answer]
I looked into this and it seems that there is no way to get the name easily but you can kind of get around it by reading every file in the current working directory and look for the name there. 

To do this you will need to turn on the node library **fs** by going to your the `bruno.json` file in your collection folder and adding this element:
```json
"scripts": {
  "filesystemAccess": {
    "allow": true
  }
 }
```
Then in your collection folder add this snippet to the post-script:
```JavaScript
const fs = require("fs");
const path = require("path");
const dir = path.join(bru.cwd());
const getAllFiles = function(dirPath, arrayOfFiles) {
  files = fs.readdirSync(dirPath);
  arrayOfFiles = arrayOfFiles || [];
  files.forEach(function(file) {
    if (fs.statSync(dirPath + "/" + file).isDirectory()) {
      arrayOfFiles = getAllFiles(dirPath + "/" + file, arrayOfFiles);
    } else {
      arrayOfFiles.push(path.join(dirPath, "/", file));
    }
  });
  return arrayOfFiles.filter((x)=> path.extname(x)=== ".bru");
};
getAllFiles(dir).forEach(function(file) {
  const content = fs.readFileSync(file,'utf8');
  let url = new RegExp("(?<=url: )"+req.getUrl());
  if (url.test(content))
  {
    console.log(path.parse(file).name);
  }
});
```
This will read every file in the collection and look for the url inside each .bru file and use it as a lookup key. If there is no string interpolation going on in the url then it will find the name and print it out in the console.

Honestly it would probably be better to just add name as a property that you can accessed using `req` because you could programmatically want to do something to a request based on its name. All the importers create a name field and transfer it over to Bruno so it doesn't seem like it would be too problematic to add. I would encourage you to make a feature request for this.

**Reply by CoenraadS** [reply]
Thanks, I made https://github.com/usebruno/bruno/issues/987

**Comment by Wyanglade** [comment]
The solution here doesn't work for me.  My collection has multiple Post requests that all have the same Url, but a different request body.

This solution will always return the name of the last request in my collection independently of which request is being executed.

**Reply by Cliffart44** [reply]
Since [v2.4.0](https://github.com/usebruno/bruno/releases/tag/v2.4.0), it has been available via `req.getName()`.
